src/Service/LinuxTracingHandler.cpp

```diff
@@ -60,6 +62,13 @@ void LinuxTracingHandler::SetupIntrospection() {
       });
 }
 
+void LinuxTracingHandler::SetupOrbitApi(const CaptureOptions& capture_options) {
```

Comment:
This should not be in `LinuxTracingHandler` but further up in `OrbitService`, closer to `CaptureServiceImpl::Capture` (probably just in that file, this is really not a lot of code after all).
Basically the idea is that now `LinuxTracingHandler` is not the "master" producer anymore, but is just one of the producers, together with the memory producer and the receivers from the Unix-socket channel (which manual instrumentation also uses).

Reply:
Sounds good, thanks for the thorough review @dpallotti . I moved that logic directly in CaptureServiceImpl::Capture, basically calling 'orbit_api::InitializeInTracee' directly, without the 'SetupOrbitApi' wrapper.